sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueuesOptions.cs

```diff
@@ -201,6 +203,21 @@ public QueueMessageEncoding MessageEncoding
             }
         }
 
+        /// <summary>
+        /// Gets or sets the JSON serialization settings to use when binding to POCOs.
+        /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
```

Comment:
I'm pattern matching what we have exposed in the ServiceBusOptions in the WebJobs Extensions for Service Bus
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs#L150
which is how we're currently exposing the Json Serializer Settings.

I can do something different here where we have our own object for the Json serializer settings.

Did we want to open an issue to change the existing WebJobs ServiceBus Json Serializer settings?

Reply:
Are these track 1 packages, or track 2 extension packages? It's hard to tell from the name, unfortunately. If track 1, then exposing `JsonSerializerSettings` directly is fine.

Reply:
This PR is for the track 2 extensions for WebJobs Extensions for Storage.

@JoshLove-msft Do you know if the [this ServiceBusOptions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs#L150) is apart of Track 1 or Track 2?

Reply:
@heaths It's track 2.

If we go about making our own object for the JsonSerializerSettings is it okay to be referencing the many Newtonsoft.Json enums that are used in that class for many of the properties?
https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializerSettings.htm
e.g.  [JsonSerializerSettings.ConstructorHandling](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_ConstructorHandling.htm) uses [Newtonsoft.Json.ConstuctorHandling](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_ConstructorHandling.htm) and so on.

It would look like this
```
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common
{
 internal class JsonSerializationSettings
    {
        /// <summary>
        /// Using the Newtonsoft.Json.JsonSerializerSettings.
        /// </summary>
        private JsonSerializerSettings _internalSettings;

        public ReferenceLoopHandling ReferenceLoopHandling
        {
            get
            {
                return _internalSettings.ReferenceLoopHandling;
            }
            set
            {
                _internalSettings.ReferenceLoopHandling = value;
            }
        }
        public MissingMemberHandling MissingMemberHandling
        {
            get
            {
                return _internalSettings.MissingMemberHandling;
            }
            set
            {
                _internalSettings.MissingMemberHandling = value;
            }
        }
        public ObjectCreationHandling ObjectCreationHandling
        {
            get
            {
                return _internalSettings.ObjectCreationHandling;
            }
            set
            {
                _internalSettings.ObjectCreationHandling = value;
            }
        }
        public int? MaxDepth
        {
            get
            {
                return _internalSettings.MaxDepth;
            }
            set
            {
                _internalSettings.MaxDepth = value;
            }
        }
        /// And so on
    }
}
```

Is that okay?

Reply:
If the extension is already referencing and exposing Newtonsoft.Json you can probably do so again here. But, if not, we need an abstraction for all types.